File: api/internal/models/application.go
package models

import (
	"gorm.io/gorm"
)

type Application struct {
	gorm.Model
	UserID             string `json:"user_id" gorm:"foreignKey:-"`
	ApplicationGroupID uint
	Url                string
	UrlToWatch         string
	HttpSuccessCode    int
	ApplicationGroup   ApplicationGroup
	Incidents          []Incident
}

func (app Application) Table() string {
	return "applications"
}

File: front/domain/entities/application.ts
import { ApplicationGroup } from "./application-group"
import { Incident } from "./Incident"

export interface Application {
    UserID: Number,
    ApplicationGroupID: Number,
    Url: string
    UrlToWatch: string
    HttpSuccessCode: Number,
    ApplicationGroup: ApplicationGroup
    Incidents: Incident[]

}